providers/mlx5/mlx5.c

```diff
@@ -609,16 +610,35 @@ static int mlx5_map_internal_clock(struct mlx5_device *mdev,
 	return 0;
 }
 
+struct ibv_cq_ex *mlx5dv_create_cq_ex(struct ibv_context *context,
+				      struct ibv_cq_init_attr_ex *cq_attr,
+				      struct mlx5dv_cq_init_attr_ex *mlx5_cq_attr)
+{
+	return ibv_vendor_create_cq_ex(context, cq_attr, mlx5_cq_attr);
+}
```

Comment:
I get the point, I just think calling some simple helpers is clearer than this indirection through yet another function pointer. There is no 'pre' step for mlx5 - it doesn't look like it needs that lock and it duplicates the wc_flags check already

```
cq = mlx5_create_cq(...);
if (cq)
    verbs_init_cq(cq);
return cq;
```